test/CMakeLists.txt

```diff
@@ -245,13 +245,39 @@ if(WITH_GTEST)
     target_link_libraries(gtest_zlib GTest::GTest)
 
     find_package(Threads)
-    if(Threads_FOUND AND NOT BASEARCH_WASM32_FOUND)
+    # TODO: Should regular C++11 threads be used when MinGW-w64 is built with pthreads?
+    if(Threads_FOUND AND NOT BASEARCH_WASM32_FOUND AND NOT MINGW)
```

Comment:
cmake sets `CMAKE_USE_PTHREADS_INIT` if pthreads are available and developer can set `CMAKE_THREAD_PREFER_PTHREAD` before calling `find_package(Threads)` to prefer pthreads over any other thread implementation.

It's also possible to explicitly check if `libpthread.a` or `libpthread.dll.a` exists to see if MinGW is built with pthreads.